src/libraries/System.Private.CoreLib/src/System/Buffer.cs

```diff
@@ -240,6 +241,39 @@ internal static void Memmove(ref byte dest, ref byte src, nuint len)
             return;
 
         MCPY05:
+            // Misaligned access via SIMD is especially expensive on ARM64 on large data.
+            // 512 is an arbitrary threshold picked for Ampere and Apple M1.
+            //
+            // TODO: Consider doing the same on x86/AMD64 for V256 and V512
+#if HAS_CUSTOM_BLOCKS && TARGET_ARM64
+            if (Vector128.IsHardwareAccelerated && len >= 512)
```

Comment:
The rest of the method does not use Vector128. It is odd to control this behavior by `Vector128.IsHardwareAccelerated`.

Reply:
IIRC @tannergooding wanted to port the memmove code to crossplatform intrinsics and I feel like getting such alignment only makes sense on platforms that do support SIMD.

Reply:
> e.g. mono interpreter

This code is not used by mono interpreter.